pkg/controller/pod.go

```diff
@@ -51,18 +55,19 @@ func (r *ReconcilePod) Reconcile(ctx context.Context, request reconcile.Request)
 	}
 
 	var pod corev1.Pod
-	klog.V(5).Infof("Reconcile pod %s.", request.Name)
 	podKey := getQualifiedName(request.Namespace, request.Name)
+	logger := r.logger.WithValues(consts.PodNameKey, request.Name)
+
 	if err := r.controllerSharedState.cachedClient.Get(ctx, request.NamespacedName, &pod); err != nil {
 		// if the pod has been deleted, remove entry from podToClaimsMap and claimToPodMap
 		if errors.IsNotFound(err) {
-			klog.V(5).Infof("Failed to reconcile pod %s. Pod was deleted.", request.Name)
+			logger.V(5).Info("No need to reconcile. Pod was deleted.")
 			if err := r.controllerSharedState.deletePod(ctx, podKey); err != nil {
 				return reconcile.Result{Requeue: true}, err
 			}
 			return reconcile.Result{}, nil
 		}
-		klog.Errorf("Error getting the pod %s. Error: %v", podKey, err)
+		r.controllerSharedState.eventRecorder.Eventf(pod.DeepCopyObject(), corev1.EventTypeWarning, consts.ClientFailedGetEvent, "Failed to get pod (%s): %v", request.NamespacedName.String(), err)
```

Comment:
I don't believe we should create an event with the error against the pod since it is not indicative of the pod's condition, but rather an error between this controller and the API server. We should just log an error here like before.